Adds ersatz validators & performance score documentation (#64)
vegaprotocol/documentation: docs/concepts/vega-chain.md

:::

### Penalties
Validating nodes that don't meet the requirements or prove to be bad actors will have rewards withheld, and a vaildator's nominators may also receive fewer (or no) rewards. 

There is no token slashing, i.e., a tokenholder cannot lose their tokens through any actions of a validator.

Read more: [How a validating node's performance is determined](/docs/concepts/vega-chain#validating-node-performance)

## Validating nodes
The Vega network is operated by a number of independent validators, who each run a node.

> There are two types of validating nodes: consensus validators and standby validators.

Author: ```suggestion
There are two types of validating nodes: consensus validators and standby validators. There can also be pending validators, which are on the waiting list to be accepted as standby validators.
```

## Consensus validating nodes
Consensus validating nodes are responsible for agreeing on the order of transactions and creating new blocks so that all nodes can agree on the state of the network.

They receive rewards based on having enough (but not too much) self-stake, as well as how many tokenholders nominate them. The tokenholders who nominate them also receive a cut of the rewards.

If a consensus validator's stake or performance is subpar, their validator score will be lowered, and that validator's node will be chosen less frequently to propose a block, because Vega feeds the voting power of each validating node to the Tendermint consensus algorithm.

This can also affect the rewards they and their nominators receive.
If a consensus validating node stops validating, or performs poorly, then a standby validator can replace it.

### How consensus validators are chosen
Consensus validators are chosen based on a range of variables, including the validator scores, and the number of available slots. If there are no empty slots, at most one validator can be changed per epoch.

> If a standby validator has a better overall performance than an existing consensus validator, then a standby validator can be promoted to replace a consensus validator.

Reviewer: _better overall performance_ - for clarity, it's not just performance, it's also stake.

## Standby validators

> Standby (also called ersatz) validators do not contribute to the chain, but are set up to join the consensus validator set if an existing validator drops off. As they don’t participate in consensus, they don’t need to be registered with a multisig contract.

Reviewer: if an existing validator drops off - not only, also if there are free slots for tendermint validators.

> As with the consensus validators, standby validators are defined based on how much self-stake and nominated stake they have.

Reviewer: are _defined_ based - not sure _defined_ is the best word here...

> As with the consensus validators, standby validators are defined based on how much self-stake and nominated stake they have.

Author: ```suggestion
As with the consensus validators, validators are determined to be standby based on how much self-stake and nominated stake they have.
```

> Standby validators, and the tokenholders who stake them, receive a share of rewards. The rewards for standby validators are calculated and penalised in the same way as consensus validators, except scaled down. How much they are scaled is based on the network parameter `network.validators.ersatz.rewardFactor`.

Reviewer: don't know to what extent you want to go into the details here, but what's scaled by this factor is the stake represented by standby validators, and consequently their reward.

> Standby validators, and the tokenholders who stake them, receive a share of rewards. The rewards for standby validators are calculated and penalised in the same way as consensus validators, except scaled down. How much they are scaled is based on the network parameter `network.validators.ersatz.rewardFactor`.

Author: ```suggestion
Standby validators, and the tokenholders who stake them, receive a share of rewards. The rewards for standby validators are calculated and penalised in the same way as consensus validators, except scaled down based on the stake they have. How much they are scaled is based on the network parameter `network.validators.ersatz.rewardFactor`.
```

:::info 
The network is set to not allow any standby validators for alpha mainnet, and the number of validators will be increased via governance as early alpha mainnet progresses.
:::

> ### How standby validators are chosen

Reviewer: I think it's worth clarifying (as you're using the term standby) that there are 3 classes, consensus, standby, and pending. Where consensus and standby are explained and are compensated, but pending are on the waiting list to be accepted as standby (hence being compensated)

If there are free slots for one or more standby validators, and there are nodes that have submitted the transaction to join (and satisfy all joining conditions), they are added as standby validators in the next epoch. 

If a node that submits the transaction to join has a higher score than the lowest scoring standby validator (scaled up by the incumbent factor), then it will become a standby validator and the lowest scoring standby validator is removed from the standby set.

> As the nodes vying for a standby spot will not have a performance record, their performance score is calculated as the average of the performance scores of all standby validators.

Reviewer: that's not true. All nodes have performance scores. For consensus validators the performance score is based on their performance in the proposing blocks. For standby validators their performance is defined by the number of heartbeats they sent out of the last 10 expected heartbeats.

> As the nodes vying for a standby spot will not have a performance record, their performance score is calculated as the average of the performance scores of all standby validators.

Author: ```suggestion

```

> Note: An inactive node that's proposing to become a validator will have a performance score of 0 and will thus be automatically excluded, regardless of their stake.

Reviewer: this is not very clear. A node that had a ranking score (=performance score x stake score) = 0, meaning they have no stake and have 0 performance score for longer than 10 epochs are removed from vega and will have to resubmit their request to become a validator. 

A node with stake is never removed from vega validator list, otherwise they or their delegators will not be able to move their stake away from the validator.

Read more: [Becoming a validator](/docs/concepts/vega-chain#becoming-a-validator)

### Moving from standby to consensus validator
A standby validator that wants to be in line for promotion to become a consensus validator needs to do the following: 

> 1. Run a non-validating node

Reviewer: (I know it's how it's written in the spec but) this should be run a validating node.

> 1. Run a non-validating node

Author: ```suggestion
1. Run a validating node
```

### How consensus validators are chosen
Consensus validators are chosen based on a range of variables, including the validator scores, and the number of available slots. If there are no empty slots, at most one validator can be changed per epoch.

If an ersatz validator has a better overall performance than an existing consensus validator, then an ersatz validator can be promoted to replace a consensus validator.

## Ersatz (standby) validators
Ersatz validators do not contribute to the chain, but are set up to join the consensus validator set if an existing validator drops off. As they don’t participate in consensus, they don’t need to be registered with a multisig contract.

As with the consensus validators, ersatz validators are defined based on how much self-stake and nominated stake they have.

Ersatz validators, and the tokenholders who stake them, receive a share of rewards. The rewards for ersatz validators are calculated and penalised in the same way as consensus validators, except scaled down. How much they are scaled is based on the network parameter `network.validators.ersatz.rewardFactor`.

> ### How ersatz validators are chosen

Author: ```suggestion
### How standby validators are chosen
```

If there are free slots for ersatz validators, and there are nodes that have submitted the transaction to join (and satisfy all joining conditions)[link to becoming a validator], they are added as ersatz validators in the next epoch. 

If a node that submits the transaction to join has a higher score than the lowest scoring ersatz validator (scaled up by the incumbent factor), then it will become an ersatz validator and the lowest scoring ersatz validator is removed from the standby set.

As the nodes vying for an ersatz spot will not have a performance record, their performance score is calculated as the average of the performance scores of all ersatz validators.

Note: An inactive node that's proposing to become a validator will have a performance score of 0 and will thus be automatically excluded, regardless of their stake.

> ### Moving from ersatz to consensus validator

Author: ```suggestion
### Moving from standby to consensus validator
```

An ersatz validator that wants to be in line for promotion to become a consensus validator needs to do the following: 

1. Run a non-validating node
2. Have enough self-stake, (as defined by the network parameter `reward.staking.delegation.minimumValidatorStake`) 
3. Forward the relevant Ethereum events

> ### Validator node performance

Author: ```suggestion
## Validator node performance
```

A validating node’s performance is calculated based on three factors: their ranking - whether they are consensus or ersatz validators, their voting power based on their performance - which defines their performance score, and their stake - which defines their validator score.

### Performance score: Consensus validators
For each block, the proposer is selected deterministically and is roughly proportional to the voting power of each consensus validator.

A validator’s performance is calculated as follows:

let `p` be the number of times the validator proposed blocks in the previous epoch 
let `b` be the number of blocks in the previous epoch 
let `v` be the voting power of the validator in the previous epoch 
let `t` be the total voting power in the previous epoch
let expected = `v*b/t` the number of blocks the validator is expected to propose. 
Then `validator_performance = max(0.05, min((p/expected, 1))`

> ### Performance score: Candidate and ersatz validators

Author: ```suggestion
### Performance score: Standby validators
```

3. Self-stake to their validator Vega key at least as much as the amount defined by the reward.staking.delegation.minimumValidatorStake network parameter
4. Wait for others to delegate to them (It would be worth announcing to the community that you have started a node and are looking for stake)

### How candidate validators are ranked
At the end of each epoch, the Vega network will calculate the unnormalised validator score (link). The consensus validators during that epoch will have their validator scores scaled by (1 + `network.validators.incumbentBonus`). This number combines self-stake and nominated stake with the performance score (which measures basic node performance).

Vega sorts all current consensus validators from the highest performance score to the lowest. All of those who submit a transaction expressing intent to be a validator are then sorted by their validator score, highest to lowest.

* If there are already empty consensus validator slots, then the non-consensus validators who have the top scores are moved in to be consensus validators, starting in the next epoch.

> * If  a potential validator has a higher score than the lowest incumbent consensus validator, then in the next epoch the higher-scoring validator becomes a consensus validator, and the lowest scoring incumbent becomes an ersatz validator.

Author: ```suggestion
* If a potential validator has a higher score than the lowest incumbent consensus validator, then in the next epoch the higher-scoring validator becomes a consensus validator, and the lowest scoring incumbent becomes an ersatz validator.
```

- Validator 1 (and its nominators) will receive 0 rewards. Validator 2 & 3 will split the 1000 equally.

## Becoming a validator
A node operator that wants to express interest in running a validating node for Vega needs to do the following: 

1. Start a Vega node as non-validating node, including the associated infrastructure: a system with a minimum of 4 cores, 16GB RAM, and 256GB SSD (subject  to increase in the future)
2. Submit a transaction using their keys, announcing they want to validate, and receive a response that the network has verified key ownership (see below)
3. Self-stake to their validator Vega key at least as much as the amount defined by the reward.staking.delegation.minimumValidatorStake network parameter
4. Wait for others to delegate to them. It would be worth announcing to the community that you have started a node and are looking for stake)

### How candidate validators are ranked

> At the end of each epoch, the Vega network will calculate the unnormalised validator score (link). The consensus validators during that epoch will have their validator scores scaled by (1 + `network.validators.incumbentBonus`). This number combines self-stake and nominated stake with the performance score (which measures basic node performance).

Author: ```suggestion
At the end of each epoch, the Vega network will calculate validator score. The consensus validators during that epoch will have their validator scores scaled by (1 + `network.validators.incumbentBonus`). This number combines self-stake and nominated stake with the performance score (which measures basic node performance).
```

- Validator 1 (and its nominators) will receive 0 rewards. Validator 2 & 3 will split the 1000 equally.

## Becoming a validator

:::info 
The network is set to not allow any standby validators for alpha mainnet, and the number of validators will be increased via governance as early alpha mainnet progresses.
:::

A node operator that wants to express interest in running a validating node for Vega needs to do the following: 

> 1. Start a Vega node as non-validating node, including the associated infrastructure: a system with a minimum of 4 cores, 16GB RAM, and 256GB SSD (subject  to increase in the future)

Reviewer: validating node

> 1. Start a Vega node as non-validating node, including the associated infrastructure: a system with a minimum of 4 cores, 16GB RAM, and 256GB SSD (subject  to increase in the future)

Author: ```suggestion
1. Start a Vega validating node, including the associated infrastructure: a system with a minimum of 4 cores, 16GB RAM, and 256GB SSD (subject  to increase in the future)
```

2. Submit a transaction using their keys, announcing they want to validate, and receive a response that the network has verified key ownership (see below)
3. Self-stake to their validator Vega key at least as much as the amount defined by the reward.staking.delegation.minimumValidatorStake network parameter

> 4. Wait for others to delegate to them. It would be worth announcing to the community that you have started a node and are looking for stake)

Reviewer: (not strictly wait)
